### PR TITLE
Added CI support in GH Actions

### LINKED ISSUE #2: Move fork to GH Actions
Since the original fork of crossguid was using Travis CI, we'll want to add basic GH Actions support to ensure that builds are at least passing.

Dependant on #1 being resolved first.

### PR DESCRIPTION
**Pull request by capnkenny**
Resolves #2

### CODE REVIEW

File: .github/workflows/build-system.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        id: build
+        run: cmake --build build --config ${{ matrix.cfg.configuration }}
+
+      - name: Install CrossGuid - ${{ matrix.cfg.configuration }}
+        if: steps.build.outcome == 'success' || steps.config2.outcome == 'success'
+        run: cmake --install build --config "${{ matrix.cfg.configuration }}" --prefix "${{ matrix.cfg.configuration }}/install"
+
+      - name: Test CrossGuid - ${{ matrix.cfg.configuration }}
+        if: steps.build.outcome == 'success' || steps.config2.outcome == 'success'
+        run: cd build/test && ctest
```
Comment (RubyNova) [inline .github/workflows/build-system.yml]: Missing EOF

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 set(SOURCES
-        Guid.cpp)
+        guid.cpp)
```
Comment (RubyNova) [inline src/CMakeLists.txt:2]: What happened here?

**Review by RubyNova — APPROVED** [review]
Looks fine bar a missing EOF

File: src/CMakeLists.txt (hunk 2 above)
Comment (capnkenny) [reply src/CMakeLists.txt:2]: Reminder from discussion earlier: Case-sensitivity in Unix-based OSes causes CMake to fail so I just had it match the source file naming.

**Pull request merged**